reference/performance.mdx

```diff
@@ -5,36 +5,288 @@ sidebarTitle: Performance
+## Summary of pitfalls
 
-Use the guidelines below to optimize datasets and queries for best performance.
+| Practice | Severity | Impact |
+| --- | --- | --- |
+| [Mixing unrelated data in datasets](#mixing-unrelated-data-in-datasets) | Critical | Combining unrelated data inflates schema, slows queries |
+| [Excessive backfilling, big difference between _time and _sysTime](#excessive-backfilling-and-large-_time-vs-_systime-gaps) | Critical | Creates overlapping blocks, breaks time-based indexing |
+| [Large number of fields in a dataset](#large-number-of-fields-in-a-dataset) | High | Very high dimensionality slows down query performance |
+| [Failing to use _time](#failing-to-use-the-_time-field-for-event-timestamps) | High | No efficient time-based filtering |
+| [Overly wide queries (project *)](#overly-wide-queries-returning-more-fields-than-needed) | High | Returns massive unneeded data |
+| [Mixed data types in the same field](#mixing-unrelated-data-in-datasets) | Moderate | Reduces compression, complicates queries |
+| [Using regex when simpler filters suffice](#regular-expressions-when-simple-filters-suffice) | Moderate | More CPU-heavy scanning |
+| [Overusing runtime JSON parsing (parse_json)](#overusing-runtime-json-parsing-parse_json) | Moderate | CPU overhead, no indexing on nested fields |
+| [Virtual fields for simple transformations](#virtual-fields-for-simple-transformations) | Low | Extra overhead for trivial conversions |
+| [Poor filter order in queries](#poor-filter-order-in-queries) | Low | Suboptimal scanning of data |
 
-## Optimize datasets
+## Mixing unrelated data in datasets
+
+### Problem
+
+A “kitchen-sink” dataset is one in which events from multiple, unrelated applications or services get lumped together, often resulting in:
+
+- **Excessive width (too many columns)**: Adding more and more unique fields bloats the schema, reducing query throughput.
+- **Mixed data types in the same field**: For example, some events store `user_id` as a string, while others store it as a number in the same `user_id` field.
+- **Unrelated schemas in a single dataset**: Columns that make sense for one app might be `null` or typed differently for another.
+
+These issues reduce compression efficiency, make indexing less effective, and force Axiom to scan more data than necessary.
+
+### Why it matters
+
+- **Slower queries**: Each query must parse wider blocks of data and handle inconsistent column types.
+- **Higher resource usage**: Wide schemas reduce row packing in blocks, harming throughput and potentially raising costs.
+- **Harder data exploration**: When fields differ drastically between events, discovering the correct columns or shaping queries becomes more difficult.
+
+### How to fix it
+
+- **Keep datasets narrowly focused:** Group data from the same application or service in its own dataset. For example, keep `k8s_logs` separate from `web_traffic`.
```

Comment:
This is a tricky example, because `k8s_logs` typically, and by definition, receive logs from all the services running on top of a kubernetes cluster. I'd suggest that for datasets where query performance is critical, splitting it off of a kitchen sink dataset is ideal, but starting off with a dataset per application is too onerous an ask, and unnecessary for low to mid scale.